### PR TITLE
Optimization of offsets for MAXAR_content_geojson string properties

### PR DESCRIPTION
**Pull request by timoore**
## Description

Offsets to string property values were always stored as 64 bit integers. This PR chooses the appropriate size, which can be UINT8, UINT16, UINT32, or UINT64

## Issue number or link

#1414 

## Author checklist

- [X] I have submitted a [Contributor License Agreement](https://github.com/CesiumGS/community/tree/main/CLAs) (only needed once).
- [X] I have done a full self-review of my code.
- [ ] I have updated `CHANGES.md` with a short summary of my change (for user-facing changes).
- [X] I have added or updated unit tests to ensure consistent code coverage as necessary.
- [ ] I have updated the documentation as necessary.

## Testing plan

Run a client with a MAXAR_content_geojson tileset and verify that string metadata is still coherent.

## Reviewer checklist

Thank you for taking the time to review this PR. By approving a PR you are taking as much responsibility for these changes as the author.

As you review, please go through the checklist below:

- [ ] Review and run all parts of the test plan on this branch and verify it matches expectations.
    - [ ] If the issue is a bug please make sure you can reproduce the bug in the main branch and then checkout this branch to make sure it actually solved the issue.
- [ ] Review the code and make sure you do not have any remaining questions or concerns. You should understand the code change and the chosen approach. If you are not confident or have doubts about the code, please do not hesitate to ask questions.
    - [ ] Code should follow the [C++ Style Guide](https://cesium.com/learn/cesium-native/ref-doc/style-guide.html)
- [ ] Review the unit tests and make sure there are no missing tests or edge cases.
- [ ] Review documentation changes and updates to `CHANGES.md` to make sure they accurately cover the work in this PR.
- [ ] Verify that the [Contributor License Agreement](https://github.com/CesiumGS/community/tree/main/CLAs) has been submitted, if needed.

### CODE REVIEW

File: CesiumVectorData/src/GltfConverter.cpp
Comment (j9liu) [inline CesiumVectorData/src/GltfConverter.cpp:1]: Update `CHANGES.md` for this with a note under "Fixes"?

File: CesiumVectorData/src/GltfConverter.cpp [hunk 1]
```diff
@@ -90,6 +90,57 @@ std::vector<uint32_t> triangulatePolygon(const Polygon& polygonIn) {
...
+  auto parameterIt = std::find_if(
+      &offsetParameters[0],
+      &offsetParameters[4],
+      [size](const PropertyOffsetParameter& param) {
+        return size == param.size;
+      });
+  // Shouldn't happen
+  if (parameterIt == &offsetParameters[4]) {
+    return offsetParameters[3].gltfConstant;
+  }
```
Comment (j9liu) [inline CesiumVectorData/src/GltfConverter.cpp]: If this shouldn't happen, perhaps indicate so with a debug / `CESIUM_ASSERT` statement instead?

File: CesiumVectorData/src/GltfConverter.cpp [hunk 2]
```diff
@@ -264,15 +315,35 @@ struct GltfConverterImpl {
     return mapIt->second;
   }
 
+  // Offsets refer to arrays, so their maximum possible size is the platform's
+  // size_t.
+  using StringOffset = size_t;
```
Comment (j9liu) [inline CesiumVectorData/src/GltfConverter.cpp]: **NIT**: I was confused by what "refer to" meant on the first read, but maybe this would be clearer?

```suggestion
  // Since offsets refer to the offset of array elements, their maximum
  // possible size is the platform's size_t.
  using StringOffset = size_t;
```

**Review by j9liu — COMMENTED** [review]
Thanks @timoore! I tried this out in Unreal and confirmed that the string data still returns correctly.

Just a few minor comments before merge.

**Review by j9liu — APPROVED** [review]
Thanks @timoore !
